src/controllers/dlgcontrollerlearning.cpp

```diff
@@ -371,6 +371,11 @@ void DlgControllerLearning::commitMapping() {
 }
 
 void DlgControllerLearning::visit(MidiController* pMidiController) {
+    // Disconnect everything so we don't end up with duplicate connections
+    // after pressing the "Learn Another" button
+    pMidiController->disconnect(this);
+    this->disconnect(pMidiController);
```

Comment:
Explicitly dereferencing this is unusual, but I suppose you did this on purpose in this special case.

Maybe rewording as "Disconnect everything in both directions..." could help to avoid any misinterpretations.

Reply:
Is the issue just the double-delivered signal? I'd suggest using Qt::UniqueConnection instead of disconnect.

Reply:
I also though about this solution. But then you need to take care of every single connection and someone might forget to add this parameter in the future. The safe solution would include both strategies, i.e. disconnect from the controller before connecting and use Qt::UniqueConnection for new connections.

Reply:
> Explicitly dereferencing this is unusual, but I suppose you did this on purpose in this special case.

Yes, it felt weird to me otherwise. Lack of symmetry :)

Are you okay to leave it like this? I've changed the comment.